core/src/main/scala/scala/pickling/spi/PicklerRegistry.scala

```diff
@@ -56,23 +56,22 @@ trait PicklerRegistry {
     *             In those situations, the pickler should be able to handle arbitrary (existential) type parameters.
     * @param p  The pickler to register.
     */
-  def registerPickler[T](key: String, p: Pickler[T]): Unit
-  def registerPickler[T](p: Pickler[T]): Unit = registerPickler(p.tag.key, p)
+  def registerPickler(key: String, p: Pickler[_]): Unit
+  def registerPickler(p: Pickler[_]): Unit = registerPickler(p.tag.key, p)
```

Comment:
This is a great full explanation, including the code snippets and references to sbt. I do have a better understanding of what you meant by "existential types" being harmful. Especially because I've also run into some hard-to-solve problems, the first being the situation when you can't prove that two existential types are the same.
  
I aimed at the least powerful abstraction, and to me that was omitting the type when it was no longer necessary and allowing any `Pickler` type. To wrap up what you've said, the problem with my approach is that it introduces a seemingly simple abstraction that leads to less composability and introduces weird compiler issues when other APIs come into play. At least, that's my general conclusion.